cmd/server.go

```diff
@@ -62,24 +72,48 @@ func serverMain(cmd *cobra.Command, args []string) exitCode {
 func addServerArgs(cmd *cobra.Command) {
 	cmd.Flags().String("loglevel", "info", "Set the logging level")
 	cmd.Flags().String("logfile", "", "Set the log file's location. Defaults to stdout")
-	cmd.Flags().String("external-plugins", "", "Specify the file to load any external plugins")
 	cmd.Flags().String("cpuprofile", "", "Write cpu profile to file")
+	cmd.Flags().String("config-file", config.DefaultFile(), "Set the config file's location")
 }
 
 func bindServerArgs(cmd *cobra.Command, args []string) {
 	// Only bind config lookup when invoking the specific command as viper bindings are global.
 	errz.Fatal(viper.BindPFlag("loglevel", cmd.Flags().Lookup("loglevel")))
 	errz.Fatal(viper.BindPFlag("logfile", cmd.Flags().Lookup("logfile")))
-	errz.Fatal(viper.BindPFlag("external-plugins", cmd.Flags().Lookup("external-plugins")))
 	errz.Fatal(viper.BindPFlag("cpuprofile", cmd.Flags().Lookup("cpuprofile")))
 }
 
-// OptsFromFlags returns server.Opts as set by command-line flags.
-func serverOptsFromFlags() server.Opts {
+// serverOptsFor returns server.Opts for the given command.
+func serverOptsFor(cmd *cobra.Command) (server.Opts, error) {
+	// Read the config
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		panic(err.Error())
+	}
+	if err := config.ReadFrom(configFile); err != nil {
+		return server.Opts{}, err
+	}
+
+	// Unmarshal the external plugins, if any are specified
+	var externalPlugins []plugin.ExternalPluginSpec
+	if externalPluginsRaw := viper.Get("external_plugins"); externalPluginsRaw != nil {
```

Comment:
Why is this underscore, unlike all the other flags?

Reply:
Oh, that's not a flag anymore. It's underscore b/c in the YAML, it'd be specified under an `external_plugins` key. Don't mind changing it to `external-plugins` though

Reply:
Can you right your YAML file without quoting it that way?